pkgs/top-level/all-packages.nix

```diff
@@ -37162,8 +37162,11 @@ with pkgs;
   elan = callPackage ../applications/science/logic/elan {};
   mathlibtools = with python3Packages; toPythonApplication mathlibtools;
 
-  leo2 = callPackage ../applications/science/logic/leo2
-    { inherit (ocaml-ng.ocamlPackages_4_05) ocaml camlp4; };
+  leo2 = callPackage ../applications/science/logic/leo2 {
+    ocamlPackages = ocaml-ng.mkOcamlPackages (ocaml-ng.ocamlPackages_4_14.ocaml.override {
+      unsafeStringSupport = true;
+    });
+  };
```

Comment:
Why is this necessary? A comment would be nice.

This smells a tad too complex for all-packages.

Reply:
The reason some packages remain on the obsolete OCaml 4.05 is that this is the last version allowing strings to be mutable. Enabling `unsafeStringSupport` often allows them to bump directly to the last versions in the 4 series, which support aarch64-darwin.

If this is too complex for all-packages, maybe it's time to introduce `ocamlPackages_4_14_unsafe_string`, as [what was done on OPAM](https://ocaml.org/p/ocaml-variants/4.11.2%2Bdefault-unsafe-string)?

Reply:
I think I'd prefer that. Also with that explanation you just wrote as a comment to explain its existence and need.

Perhaps ocaml 4.05 should probably be aliased to `ocamlPackages_4_14_unsafe_string` on aarch64-darwin.